job-service/src/main/java/com/hpe/caf/services/job/jobtype/DefaultDefinitionParser.java

```diff
@@ -0,0 +1,250 @@
+            }
+            return taskPipe;
+        }
+
+        public String getTargetPipe(final AppConfig appConfig)
+            throws InvalidJobTypeDefinitionException
+        {
+            final String targetPipe = appConfig.getJobTypeProperty(getId(), "target_pipe");
+            if (targetPipe == null) {
+                throw new InvalidJobTypeDefinitionException(
+                    getId() + ": target pipe is not configured");
+            }
+            return targetPipe;
+        }
+
+        public void setConfigurationProperties(
+            final List<ConfigurationProperty> configurationProperties
+        ) {
+            this.configurationProperties = configurationProperties;
+        }
+
+        public Map<String, String> getConfiguration(final AppConfig appConfig)
+            throws InvalidJobTypeDefinitionException
+        {
+            final List<ConfigurationProperty> properties = configurationProperties == null ?
+                Collections.emptyList() : configurationProperties;
+
+            final Map<String, String> configuration = new HashMap<>();
+            for (final ConfigurationProperty property : properties) {
+                final String propertyName = property.getName(this);
+                final String propertyValue = appConfig.getJobTypeProperty(id, propertyName);
+                if (propertyValue == null) {
+                    throw new InvalidJobTypeDefinitionException(
+                        getId() + ": configuration is not available: " + propertyName);
+                }
+                // preserve property name case in the output, even though the configuration lookup
+                // ignores it
+                configuration.put(propertyName, propertyValue);
+            }
+
+            return configuration;
+        }
+
+        public void setJobParametersSchema(final Object jobParametersSchema) {
+            this.jobParametersSchema = jobParametersSchema;
+        }
+
+        public JsonNode getParametersSchema() {
+            // should never fail
+            return objectMapper.convertValue(
+                jobParametersSchema == null ? DEFAULT_JOB_PARAMETERS_SCHEMA : jobParametersSchema,
+                JsonNode.class);
+        }
+
+        public void setTaskDataScript(final String taskDataScript) {
+            this.taskDataScript = taskDataScript;
+        }
+
+        public String getTaskDataScript() throws InvalidJobTypeDefinitionException {
+            if (taskDataScript == null) {
+                throw new InvalidJobTypeDefinitionException(
+                    getId() + ": missing property: taskDataScript");
+            }
+            return taskDataScript;
+        }
+
+    }
+
+
+    /**
+     * An item in a definition's `configurationProperties`, parsed from yaml.
+     */
+    public static final class ConfigurationProperty {
+        private String name;
+        private String description;
```

Comment:
I'm not sure you would ever care what description is - the field's only there because the parser fails when it finds a YAML property and nowhere to put the value.  But there's no harm adding it for completeness.

Reply:
Ok.  Either is fine.